lib/ansible/modules/storage/netapp/na_ontap_volume.py

```diff
@@ -0,0 +1,614 @@
+        try:
+            self.cluster.invoke_successfully(volume_move,
+                                             enable_tunneling=True)
+            self.ems_log_event("move")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error moving volume %s: %s'
+                                  % (self.name, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def rename_volume(self):
+        """
+        Rename the volume.
+
+        Note: 'is_infinite' needs to be set to True in order to rename an
+        Infinite Volume.
+        """
+        if self.is_infinite:
+            volume_rename = netapp_utils.zapi\
+                .NaElement.create_node_with_children(
+                    'volume-rename-async',
+                    **{'volume-name': self.name, 'new-volume-name': str(
+                        self.new_name)})
+        else:
+            volume_rename = netapp_utils.zapi\
+                .NaElement.create_node_with_children(
+                    'volume-rename', **{'volume': self.name,
+                                        'new-volume-name': str(self.new_name)})
+        try:
+            self.server.invoke_successfully(volume_rename,
+                                            enable_tunneling=True)
+            self.ems_log_event("rename")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error renaming volume %s: %s'
+                                  % (self.name, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def resize_volume(self):
+        """
+        Re-size the volume.
+
+        Note: 'is_infinite' needs to be set to True in order to rename an
+        Infinite Volume.
+        """
+        if self.is_infinite:
+            volume_resize = netapp_utils.zapi\
+                .NaElement.create_node_with_children(
+                    'volume-size-async',
+                    **{'volume-name': self.name, 'new-size': str(
+                        self.size)})
+        else:
+            volume_resize = netapp_utils.zapi\
+                .NaElement.create_node_with_children(
+                    'volume-size', **{'volume': self.name, 'new-size': str(
+                        self.size)})
+        try:
+            self.server.invoke_successfully(volume_resize,
+                                            enable_tunneling=True)
+            self.ems_log_event("resize")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error re-sizing volume %s: %s'
+                                  % (self.name, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def change_volume_state(self):
+        """
+        Change volume's state (offline/online).
+
+        Note: 'is_infinite' needs to be set to True in order to change the
```

Comment:
Not possible to glean the volume type from the result of get_volume and choose the right operation automatically? If not, these admonitions need to be in the docs, not in the code...